src/common/device/AndroidDeviceManager.ts

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import fs from 'node:fs';
+import path from 'node:path';
+import { Logger } from '@salesforce/core';
+import { AndroidUtils } from '../AndroidUtils.js';
+import { CaseInsensitiveStringMap, Version } from '../Common.js';
+import { CommonUtils } from '../CommonUtils.js';
+import { PlatformConfig } from '../PlatformConfig.js';
+import { AndroidDevice, AndroidOSType } from './AndroidDevice.js';
+import { DeviceType } from './BaseDevice.js';
+
+export class AndroidDeviceManager {
+    private logger?: Logger;
+
+    public constructor(logger?: Logger) {
+        this.logger = logger;
+    }
+
+    /**
+     * Runs the command `avdmanager list avd` then parses the output to enumerate a list of available devices.
+     * By using `osFilters` you can further filter the types of devices that you would like to be included in the results.
+     *
+     * @param osFilters Array of `{ osType: string; minOSVersion: Version }` objects which can be used to further
+     * filter the results and only include the devices that match the provided filters. Defaults to only include
+     * devices running on Google APIs 23 or later.
+     *
+     * For example, to include various OS types and versions you can provide:
+     * ```
+     * [
+     *   { osType: AndroidOSType.googleAPIs, minOSVersion: new Version(33, 0, 0) }, // Google APIs 33 or newer
+     *   { osType: AndroidOSType.googlePlayStore, minOSVersion: new Version(34, 0, 0) }, // Google APIs Play Store 34 or newer
+     *   { osType: AndroidOSType.androidWear, minOSVersion: new Version(3, 0, 0) }, // Android Wear 3 or newer
+     * ]
+     * ```
+     *
+     * You can also pass in `null` for osFilters in order to skip filtering and include all discovered devices.
+     *
+     * @returns An array of available devices filtered using the `osFilter` parameter.
+     */
+    public async enumerateDevices(
```

Comment:
With my new changes, our code can now handle devices of different types (phone, tablet, tv, auto, watch, vr, ...) But of course for now we only want to deal with phone/tablet so I've made the default values of the filters such that it would preserve the old behavior for backwards compatibility.